src/components/BulkEnrollmentPage/CourseSearchResults.jsx

```diff
@@ -23,43 +30,58 @@ export const TABLE_HEADERS = {
   enroll: '',
 };
 
-export const EnrollButton = ({ row, setSelectedCourseRuns, setModalOpen }) => {
-  const handleClick = useMemo(() => () => {
-    setSelectedCourseRuns([row.original?.advertised_course_run?.key]);
-    setModalOpen(true);
-  }, [setSelectedCourseRuns, setModalOpen, row.original?.advertised_course_run?.key]);
+const AddCoursesSelectionStatus = (props) => {
+  const { courses: [selectedCourses, dispatch] } = useContext(BulkEnrollContext);
+
+  return <BaseSelectionStatus selectedRows={selectedCourses} dispatch={dispatch} {...props} />;
+};
+
+const SelectWithContext = (props) => <BaseSelectWithContext contextKey="courses" {...props} />;
+
+const SelectWithContextHeader = (props) => <BaseSelectWithContextHeader contextKey="courses" {...props} />;
+
+const selectColumn = {
+  id: 'selection',
+  Header: SelectWithContextHeader,
+  Cell: SelectWithContext,
+  disableSortBy: true,
+};
+
+export const EnrollButton = ({ row, goToNextStep, dispatch }) => {
+  const handleClick = () => {
+    dispatch(setSelectedRowsAction([row]));
+    goToNextStep();
+  };
+
   return (
     <Button
       className="enroll-button"
       variant="link"
       onClick={handleClick}
+      data-testid="tableEnrollButton"
```

Comment:
might be nice to have consistency in casing patterns for these ID's. I used/saw kebab-case elsewhere in this PR.